File: src/ProjectR/content/scripts/afflictions/Passives/DarkPortal.lua
function GetName()
  return "Dark Portal"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { TurnTriggered }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == TurnTriggered and Roll(0, 99) < 20 then
    SummonMinionCopy(char, "Imp")
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/DarkPortal.cs
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class DarkPortal : Affliction
    {
        public override string Name { get { return "Dark Portal"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TurnTriggered, 
                };
            }
        }

        protected override void OnTurnTriggered(ICharacter character)
        {
            if (RollPercentage(20))
            {
                SummonMinionCopy(character, "Imp");
            }
        }
    }
}